libs/features/placeholders.js

```diff
@@ -1,3 +1,5 @@
+import { fetchMaybeCache } from '../utils/utils.js';
```

Comment:
Couldn't this lead to some circular dependency issues? `utils.js` imports `placeholders.js` (and `icons.js` actually), which then import a method from `utils.js`. I was assuming that's the reason for not having gone this single method path in the first place. Might be worth checking that there are no undesirable side-effects to this.

Reply:
oh wow, THANK YOU. This was the root of the issue I was having with unit tests. Did not realize there were dynamic imports for them buried in `utils.js`.

For now I've moved it to `/libs/utils/helpers.js`... hopefully that's fine 🤷